### PR TITLE
Diagonal Assembly Clarification for AtPoints

### PR DESCRIPTION
**Pull request by jeremylt**
I changed this MR to clean up some internal logic and then we can do the diagonal assembly later if it becomes important

### CODE REVIEW

**Comment by jeremylt** [comment]
Note - CI failure is unrelated, this is fine to merge now

File: interface/ceed-preconditioning.c [hunk 1]
```diff
@@ -187,7 +187,8 @@ static int CeedOperatorCreateFallback(CeedOperator op) {
 
   @ref Developer
 **/
-static inline int CeedSingleOperatorAssembleAddDiagonal_Core(CeedOperator op, CeedRequest *request, const bool is_point_block, CeedVector assembled) {
+static inline int CeedSingleOperatorLinearAssembleAddDiagonal_Mesh(CeedOperator op, CeedRequest *request, const bool is_point_block,
+                                                                   CeedVector assembled) {
```
Comment (jrwrigh) [inline interface/ceed-preconditioning.c:191]: I don't like `Mesh` as a name for "not AtPoints" for some reason, but I also can't think of a better name, nor am I familiar with the relationship between AtPoints and the "normal" operators.

File: interface/ceed-preconditioning.c [hunk 2]
```diff
@@ -361,6 +362,30 @@ static inline int CeedSingleOperatorAssembleAddDiagonal_Core(CeedOperator op, Ce
...
+  @ref Developer
+**/
+static inline int CeedSingleOperatorLinearAssembleAddDiagonal(CeedOperator op, CeedRequest *request, const bool is_point_block,
+                                                              CeedVector assembled) {
+  Ceed ceed;
+  bool is_at_points;
+
+  CeedCall(CeedOperatorGetCeed(op, &ceed));
+  CeedCall(CeedOperatorIsAtPoints(op, &is_at_points));
+  CeedCheck(!is_at_points, ceed, CEED_ERROR_UNSUPPORTED, "Composite operator not supported");
```
Comment (jrwrigh) [inline interface/ceed-preconditioning.c:384]: Should this read:
```suggestion
  CeedCheck(!is_at_points, ceed, CEED_ERROR_UNSUPPORTED, "AtPoints operator not supported");
```
?

**Review by jrwrigh — APPROVED** [review]

**Comment by jrwrigh** [comment]
I think you need to rebase onto main to get the DMPlexFilter corrections.

File: interface/ceed-preconditioning.c (hunk 1 above)
Comment (jeremylt) [reply interface/ceed-preconditioning.c:191]: Yeah, I don't love it, but I don't have a better name either

Comment (jrwrigh) [reply interface/ceed-preconditioning.c:191]: We can always change it later if we think of something better.

**Pull request merged**